vmdb/lib/vmdb/util.rb

```diff
@@ -1,16 +1,15 @@
 module VMDB
   module Util
     # load all subclasses recursively
-    def self.eager_load_subclasses(klass)
+    def self.eager_load_subclasses(klass, pattern = nil)
+      pattern ||= "#{klass.underscore}_*.rb"
       ActiveSupport::Dependencies.autoload_paths.each do |root|
-        Dir.glob(File.join(root, "#{klass.underscore}_*.rb")).sort.each do |file|
-          name = File.basename(file, '.*')
-          if name =~ /^#{klass.underscore}_[^_]*$/  # filter out sub-subclasses
-            require_dependency file
-          end
+        Dir.glob(File.join(root, pattern)).each do |file|
+          parent_class = File.read(file).lines.detect { |l| l.start_with?("class ") }.to_s.split[3]
```

Comment:
Maybe a regex is better? Something like `/^(?:class|module)\s+(\w+)\s*<\s*(\w+)/`.